core/server/data/migrations/versions/3.22/05-migrate-members-subscription-settings.js

```diff
@@ -0,0 +1,142 @@
+const logging = require('../../../../../shared/logging');
+const ObjectId = require('bson-objectid').default;
+
+module.exports = {
+    config: {
+        transaction: true
+    },
+
+    async up(config) {
+        const knex = config.transacting;
+        const membersSubscriptionSettingsJSON = await knex('settings')
+            .select('value')
+            .where('key', 'members_subscription_settings')
+            .first();
+
+        const membersSubscriptionSettings = JSON.parse(membersSubscriptionSettingsJSON.value);
+
+        const membersFromAddress = membersSubscriptionSettings.fromAddress;
+        const membersAllowSelfSignup = membersSubscriptionSettings.allowSelfSignup;
+
+        const stripe = membersSubscriptionSettings.paymentProcessors[0];
+
+        const stripeDirectSecretKey = stripe.config.secret_token;
+        const stripeDirectPublishableKey = stripe.config.public_token;
+        const stripeProductName = stripe.config.product.name;
+
+        const stripePlans = stripe.config.plans;
+
+        const now = knex.raw('CURRENT_TIMESTAMP');
+
+        const operations = [{
+            key: 'members_from_address',
+            value: membersFromAddress
+        }, {
+            key: 'members_allow_signup',
+            value: membersAllowSelfSignup
```

Comment:
@allouis The boolean setting values in DB need to be stringified before as otherwise since we use TEXT field for values, it gets converted to "1" or "0" for true/false. 

We usually handle it in model layer [here](https://github.com/TryGhost/Ghost/blob/91e3630f3647e6bdfc24eba8b2ec4c1d29866ab7/core/server/models/settings.js#L134-L140), but in this case its bypassed so DB is storing it as "1" / "0" right now :) 

Changing to `membersAllowSelfSignup.toString()` should do it.